swayidle/main.c

```diff
@@ -0,0 +1,448 @@
+	wlr_log(L_DEBUG, "active state");
+	if (cmd && cmd->resume_cmd && cmd->resume_cmd->callback) {
+		cmd->resume_cmd->callback(cmd->resume_cmd->param);
+	}
+}
+
+static const struct org_kde_kwin_idle_timeout_listener idle_timer_listener = {
+	.idle = handle_idle,
+	.resumed = handle_resume,
+};
+
+struct swayidle_cmd *parse_command(int argc, char **argv) {
+	if (argc < 2) {
+		wlr_log(L_ERROR, "Too few parameters for command in parse_command");
+		return NULL;
+	}
+
+	struct swayidle_cmd *cmd = calloc(1, sizeof(struct swayidle_cmd));
+	wlr_log(L_DEBUG, "Command: %s %s", argv[0], argv[1]);
+	if (!strcmp("exec", argv[0])) {
+		cmd->callback = cmd_exec;
+	} else {
+		wlr_log(L_ERROR, "Bad command should be exec");
+		return NULL;
+	}
+	cmd->param = argv[1];
+	return cmd;
+}
+
+int parse_timeout(int argc, char **argv) {
+	if (argc < 4) {
+		wlr_log(L_ERROR, "Too few parameters to timeout command. \
+Usage: timeout <seconds> exec <command>");
+		exit(-1);
+	}
+	errno = 0;
+	char *endptr;
+	int seconds = strtoul(argv[1], &endptr, 10);
+	if (errno != 0 || *endptr != '\0') {
+		wlr_log(L_ERROR, "Invalid timeout parameter '%s', it should be a \
+numeric value representing seconds", optarg);
+		exit(-1);
+	}
+	struct swayidle_timeout_cmd *cmd =
+		calloc(1, sizeof(struct swayidle_timeout_cmd));
+	cmd->timeout = seconds*1000;
+
+	wlr_log(L_DEBUG, "Register idle timeout at %d ms", cmd->timeout);
+	wlr_log(L_DEBUG, "Setup idle");
+	cmd->idle_cmd = parse_command(argc - 2, &argv[2]);
+
+	int result = 4;
+	if (argc >= 7 && !strcmp("resume", argv[4])) {
+		wlr_log(L_DEBUG, "Setup resume");
+		cmd->resume_cmd = parse_command(argc - 5, &argv[5]);
+		result = 7;
+	}
+	list_add(state.timeout_cmds, cmd);
+	return result;
+}
+
+int parse_args(int argc, char *argv[]) {
+	int c;
+	char *endptr;
+	static uint32_t idle_timeout = 0, lock_timeout = 0;
+	char *lock_timeout_str = NULL;
+	char *idle_timeout_str = NULL;
+
+	while ((c = getopt(argc, argv, "i:hl:s:dx:")) != -1) {
+		switch(c) {
+		case 'i':
+			errno = 0;
+			idle_timeout = strtoul(optarg, &endptr, 10);
+			if (errno != 0 || *endptr != '\0') {
+				wlr_log(L_ERROR, "Invalid idle parameter '%s', it should be a \
+numeric value representing seconds", optarg);
+				exit(-1);
+			}
+			idle_timeout_str = optarg;
+			break;
+		case 'l':
+			errno = 0;
+			lock_timeout = strtoul(optarg, &endptr, 10);
+			if (errno != 0 || *endptr != '\0') {
+				wlr_log(L_ERROR, "Invalid lock parameter '%s', it should be a \
+numeric value representing seconds", optarg);
+				exit(-1);
+			}
+			lock_timeout_str = optarg;
+			break;
+		case 'd':
+			debug = true;
+			break;
+		case 'x':
+			lock_cmd = strdup(optarg);
+			break;
+		case 'h':
+		case '?':
+			printf("Usage: %s [OPTIONS]\n", argv[0]);
+			printf("  -i seconds optional\t\tidle timeout in seconds\n");
+			printf("  -l seconds optional\t\tlock timeout in seconds\n");
+			printf("  -x lock_command\t\t to raise the lockscreen on systemd \
```

Comment:
This looks weird in my terminal, can you align with spaces instead

Reply:
This is fixed